### ISSUE TITLE
Index Error Saving a Record after Upgrading from 2.x to 3.0.8

### ISSUE THREAD

**Issue by mmacfadden** [orientechnologies/orientdb#8576]
### OS: RHEL

When calling OElement.save() we are seeing the below error. This was in a database that was originally created with 2.2.x. We recently upgraded to 3.0.8 and are now seeing this in several instances.

```
Exception `7474E0BA` in storage `plocal:/orientdb/databases/6335366053283600827`: 3.0.8 - Veloce (build ace6fe7e50f069968d2ec61678848606703db1c0, branch 3.0.x)
com.orientechnologies.orient.core.exception.OStorageException: Cannot put key value entry in index
	DB name="6335366053283600827"
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doUpdateIndexEntry(OAbstractPaginatedStorage.java:2784)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.updateIndexEntry(OAbstractPaginatedStorage.java:2669)
	at com.orientechnologies.orient.core.index.OIndexMultiValues.put(OIndexMultiValues.java:173)
	at com.orientechnologies.orient.core.index.OIndexMultiValues.put(OIndexMultiValues.java:52)
	at com.orientechnologies.orient.core.index.OIndexAbstract.putInSnapshot(OIndexAbstract.java:861)
	at com.orientechnologies.orient.core.index.OIndexAbstract.applyIndexTxEntry(OIndexAbstract.java:704)
	at com.orientechnologies.orient.core.index.OIndexAbstract.addTxOperation(OIndexAbstract.java:676)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commitIndexes(OAbstractPaginatedStorage.java:2230)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:2150)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:1984)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.internalCommit(ODatabaseDocumentAbstract.java:2750)
	at com.orientechnologies.orient.core.storage.impl.local.OMicroTransaction.doCommit(OMicroTransaction.java:363)
	at com.orientechnologies.orient.core.storage.impl.local.OMicroTransaction.commit(OMicroTransaction.java:166)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.endMicroTransaction(ODatabaseDocumentEmbedded.java:756)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.executeSaveRecord(ODatabaseDocumentEmbedded.java:746)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveNew(OTransactionNoTx.java:244)
	at com.orientechnologies.orient.core.tx.OTransactionNoTx.saveRecord(OTransactionNoTx.java:174)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.saveInternal(ODatabaseDocumentAbstract.java:2076)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:2041)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:1933)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentAbstract.save(ODatabaseDocumentAbstract.java:84)
	at com.orientechnologies.orient.server.OConnectionBinaryExecutor.executeCreateRecord(OConnectionBinaryExecutor.java:374)
	at com.orientechnologies.orient.client.remote.message.OCreateRecordRequest.execute(OCreateRecordRequest.java:118)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.sessionRequest(ONetworkProtocolBinary.java:304)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.execute(ONetworkProtocolBinary.java:206)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Error during index update
	DB name="6335366053283600827"
	at com.orientechnologies.orient.core.storage.index.hashindex.local.OLocalHashTable.put(OLocalHashTable.java:1403)
	at com.orientechnologies.orient.core.storage.index.hashindex.local.OLocalHashTable.put(OLocalHashTable.java:385)
	at com.orientechnologies.orient.core.storage.index.engine.OHashTableIndexEngine.put(OHashTableIndexEngine.java:176)
	at com.orientechnologies.orient.core.storage.index.engine.OHashTableIndexEngine.update(OHashTableIndexEngine.java:184)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.doUpdateIndexEntry(OAbstractPaginatedStorage.java:2770)
	... 25 more
Caused by: java.lang.ClassCastException: com.orientechnologies.orient.core.storage.ridbag.sbtree.OMixedIndexRIDContainer cannot be cast to com.orientechnologies.orient.core.storage.ridbag.sbtree.OIndexRIDContainer
	at com.orientechnologies.orient.core.serialization.serializer.stream.OStreamSerializerSBTreeIndexRIDContainer.getObjectSize(OStreamSerializerSBTreeIndexRIDContainer.java:42)
	at com.orientechnologies.orient.core.storage.index.hashindex.local.OHashIndexBucket.addEntry(OHashIndexBucket.java:305)
	at com.orientechnologies.orient.core.storage.index.hashindex.local.OLocalHashTable.doPut(OLocalHashTable.java:1497)
	at com.orientechnologies.orient.core.storage.index.hashindex.local.OLocalHashTable.put(OLocalHashTable.java:1395)
	... 29 more

```

**Comment by mmacfadden** [comment]
I should add that rebuilding all indices seems to fix the issue.

**Comment by laa** [comment]
Fixed in 3.0.9
